client/src/components/Collections/common/CollectionEditView.vue

```diff
@@ -0,0 +1,260 @@
+                >
+                    {{ genome.text }}
+                    <!-- <template slot="afterList">
+                        <div v-observe-visibility="reachedEndOfList" v-if="hasMorePages">
+                            <span class="spinner fa fa-spinner fa-spin fa-1x" />
+                        </div>
+                    </template> -->
+                </multiselect>
+            </b-tab>
+            <b-tab>
+                <template v-slot:title> <i class="fa fa-gear"></i>{{ l(" Convert") }}</template>
+                <b>{{ l("Datatype: ") }}</b> <i>{{ datatypeFromElements }}</i>
+            </b-tab>
+            <b-tab>
+                <template v-slot:title> <i class="fa fa-database"></i>{{ l(" Datatype") }}</template>
+                <div class="alert alert-secondary" role="alert">
+                    <div class="float-left">Change datatype of all elements in collection</div>
+                    <div class="text-right">
+                        <button
+                            class="save-collection-edit btn btn-primary"
+                            @click="clickedSave('file_ext', extension)"
+                            :disabled="extension.id == datatypeFromElements"
+                        >
+                            {{ l("Save") }}
+                        </button>
+                    </div>
+                </div>
+                <b>{{ l("Datatype: ") }}</b>
+                <multiselect
+                    v-model="extension"
+                    deselect-label="Can't remove this value"
+                    track-by="id"
+                    label="text"
+                    :options="extensions"
+                    :searchable="true"
+                    :allow-empty="false"
+                >
+                    {{ extension.text }}
+                    <!-- <template slot="afterList">
+                        <div v-observe-visibility="reachedEndOfList" v-if="hasMorePages">
+                            <span class="spinner fa fa-spinner fa-spin fa-1x" />
+                        </div>
+                    </template> --> </multiselect
+                ><i>original input: {{ datatypeFromElements }}</i>
+            </b-tab>
+            <b-tab>
+                <template v-slot:title> <i class="fa fa-user"></i>{{ l(" Permissions") }}</template>
+                <p>WIP Permissions</p>
+            </b-tab>
+        </b-tabs>
+    </div>
+</template>
+
+<script>
+import Vue from "vue";
+import BootstrapVue from "bootstrap-vue";
+import axios from "axios";
+import { prependPath } from "utils/redirect";
+import UploadUtils from "mvc/upload/upload-utils";
+import _l from "utils/localization";
+import Multiselect from "vue-multiselect";
+import { errorMessageAsString } from "utils/simple-error";
+//import VueObserveVisibility from "vue-observe-visibility";
+
+//Vue.use(VueObserveVisibility);
+Vue.use(BootstrapVue);
+export default {
+    created() {
+        this.apiCallToGetData();
```

Comment:
You could reuse the `DatasetCollectionProvider` in client/src/components/WorkflowInvocationState/providers, and I would refactor the view so that you only need to make a single call to the API.

Reply:
I'll use this whenever I implement #6966; for now, I took out the call for all the collection data, since it's not necessary yet.